src/plugins/intel_cpu/tests/unit/transformations/arm/grid_sample_decomposition.cpp

```diff
@@ -0,0 +1,417 @@
+// Copyright (C) 2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <gtest/gtest.h>
+
+#include <memory>
+#include <string>
+#include <tuple>
+#include <vector>
+
+#include "common_test_utils/ov_test_utils.hpp"
+#include "openvino/core/model.hpp"
+#include "openvino/op/constant.hpp"
+#include "openvino/op/gather_nd.hpp"
+#include "openvino/op/grid_sample.hpp"
+#include "openvino/op/parameter.hpp"
+#include "openvino/op/result.hpp"
+#include "openvino/op/transpose.hpp"
+#include "openvino/pass/manager.hpp"
+#include "transformations/cpu_opset/arm/pass/grid_sample_decomposition.hpp"
+#include "transformations/init_node_info.hpp"
+
+using namespace testing;
+using namespace ov::intel_cpu;
+
+// ========== Helper function declarations ==========
+std::shared_ptr<ov::Model> create_expected_decomposed_pattern(
+    const ov::PartialShape& data_shape,
+    const ov::PartialShape& grid_shape,
+    const ov::element::Type& data_type,
+    const ov::element::Type& grid_type,
+    const ov::op::v9::GridSample::Attributes& attrs);
+
+// ========== Helper function implementations ==========
+std::shared_ptr<ov::Model> create_expected_decomposed_pattern(
+    const ov::PartialShape& data_shape,
+    const ov::PartialShape& grid_shape,
+    const ov::element::Type& data_type,
+    const ov::element::Type& grid_type,
+    const ov::op::v9::GridSample::Attributes& attrs) {
+
+    auto data = std::make_shared<ov::op::v0::Parameter>(data_type, data_shape);
+    auto grid = std::make_shared<ov::op::v0::Parameter>(grid_type, grid_shape);
+
+    // This is where we would manually construct the expected decomposed pattern
+    // For now, we'll use the transformation to generate it, but ideally we'd build
+    // the exact pattern we expect: GatherND, Transpose, and other ops
+
+    auto grid_sample = std::make_shared<ov::op::v9::GridSample>(data, grid, attrs);
+    auto result = std::make_shared<ov::op::v0::Result>(grid_sample);
+    auto model = std::make_shared<ov::Model>(ov::ResultVector{result}, ov::ParameterVector{data, grid});
+
+    // Apply transformation to get the decomposed pattern
+    ov::pass::Manager manager;
+    manager.register_pass<ov::intel_cpu::GridSampleDecomposition>();
+    manager.run_passes(model);
```

Comment:
The same concern as in [this discussion](https://github.com/openvinotoolkit/openvino/pull/31544#discussion_r2285244754)